spark/src/test/scala/org/apache/spark/sql/delta/perf/OptimizeMetadataOnlyDeltaQuerySuite.scala

```diff
@@ -17,141 +17,594 @@
+    dfPart2 = generateRowsDataFrame(spark.range(6L, 11L))
 
     withSQLConf(DeltaSQLConf.DELTA_COLLECT_STATS.key -> "false") {
-      df.write.format("delta").mode(SaveMode.Overwrite).saveAsTable(noStatsTableName)
-      df.write.format("delta").mode(SaveMode.Overwrite).saveAsTable(mixedStatsTableName)
+      dfPart1.write.format("delta").mode(SaveMode.Overwrite).saveAsTable(noStatsTableName)
+      dfPart1.write.format("delta").mode(SaveMode.Overwrite).saveAsTable(mixedStatsTableName)
 
       spark.sql(s"DELETE FROM $noStatsTableName WHERE id = 1")
       spark.sql(s"DELETE FROM $mixedStatsTableName WHERE id = 1")
 
-      df2.write.format("delta").mode("append").saveAsTable(noStatsTableName)
+      dfPart2.write.format("delta").mode("append").saveAsTable(noStatsTableName)
     }
 
     withSQLConf(DeltaSQLConf.DELTA_COLLECT_STATS.key -> "true") {
-      import io.delta.tables._
-
-      df.write.format("delta").mode(SaveMode.Overwrite).saveAsTable(testTableName)
-      df.write.format("delta").mode(SaveMode.Overwrite).save(testTablePath)
+      dfPart1.write.format("delta").mode(SaveMode.Overwrite).saveAsTable(testTableName)
 
       spark.sql(s"DELETE FROM $testTableName WHERE id = 1")
-      DeltaTable.forPath(spark, testTablePath).delete("id = 1")
 
-      df2.write.format("delta").mode(SaveMode.Append).saveAsTable(testTableName)
-      df2.write.format("delta").mode(SaveMode.Append).save(testTablePath)
-      df2.write.format("delta").mode(SaveMode.Append).saveAsTable(mixedStatsTableName)
+      dfPart2.write.format("delta").mode(SaveMode.Append).saveAsTable(testTableName)
+      dfPart2.write.format("delta").mode(SaveMode.Append).saveAsTable(mixedStatsTableName)
+
+      // Run updates to generate more Delta Log and trigger a checkpoint
+      // and make sure stats works after checkpoints
+      for (a <- 1 to 10) {
+        spark.sql(s"UPDATE $testTableName SET data='$a' WHERE id = 7")
+      }
+      spark.sql(s"UPDATE $testTableName SET data=NULL WHERE id = 7")
+
+      // Creates an empty (numRecords == 0) AddFile record
+      generateRowsDataFrame(spark.range(11L, 12L))
+        .write.format("delta").mode("append").saveAsTable(testTableName)
+      spark.sql(s"DELETE FROM $testTableName WHERE id = 11")
     }
   }
 
-  test("Select Count: basic") {
+  test("count - simple query") {
+    val expectedPlan = "LocalRelation [none#0L]"
+
     checkResultsAndOptimizedPlan(
       s"SELECT COUNT(*) FROM $testTableName",
-      Seq(Row(totalRows)),
-      "LocalRelation [none#0L]")
-  }
+      expectedPlan)
 
-  test("Select Count: column alias") {
     checkResultsAndOptimizedPlan(
-      s"SELECT COUNT(*) as MyColumn FROM $testTableName",
-      Seq(Row(totalRows)),
-      "LocalRelation [none#0L]")
+      () => spark.read.format("delta").table(testTableName)
+        .agg(count(col("*"))),
+      expectedPlan)
   }
 
-  test("Select Count: table alias") {
+  test("min-max - simple query") {
```

Comment:
is it possible to create a test matrix? that way we can see what is covered and what is not. There are too many cases.

Reply:
Done